ui/src/components/PWABadge.vue

```diff
@@ -0,0 +1,118 @@
+<script setup>
+// eslint-disable-next-line import/no-unresolved
+import { useRegisterSW } from 'virtual:pwa-register/vue'
+import { computed, ref } from 'vue'
+
+// check for updates every hour
+const period = 60 * 60 * 1000
+
+const swActivated = ref(false)
+
+/**
+ * This function will register a periodic sync check every hour, you can modify the interval as needed.
+ * @param {string} swUrl
+ * @param {ServiceWorkerRegistration} r
+ */
+function registerPeriodicSync (swUrl, r) {
+    if (period <= 0) return
+
+    setInterval(async () => {
+        if ('onLine' in navigator && !navigator.onLine) { return }
+
+        const resp = await fetch(swUrl, {
+            cache: 'no-store',
+            headers: {
+                cache: 'no-store',
+                'cache-control': 'no-cache'
+            }
+        })
+
+        if (resp?.status === 200) { await r.update() }
+    }, period)
+}
+
+const { offlineReady, needRefresh, updateServiceWorker } = useRegisterSW({
+    immediate: true,
+    onRegisteredSW (swUrl, r) {
+        if (period <= 0) return
+        if (r?.active?.state === 'activated') {
+            swActivated.value = true
+            registerPeriodicSync(period, swUrl, r)
+        } else if (r?.installing) {
+            r.installing.addEventListener('statechange', (e) => {
+                /** @type {ServiceWorker} */
+                const sw = e.target
+                swActivated.value = sw.state === 'activated'
+                if (swActivated.value) { registerPeriodicSync(period, swUrl, r) }
+            })
+        }
+    }
+})
+
+const title = computed(() => {
+    if (offlineReady.value) { return 'App is offline ready' }
+    if (needRefresh.value) { return 'New content available, click on reload button to update.' }
+    return ''
+})
+
+function close () {
+    offlineReady.value = false
+    needRefresh.value = false
+}
+</script>
+
+<template>
+    <div
+        v-if="offlineReady || needRefresh"
+        class="pwa-toast"
+        aria-labelledby="toast-message"
+        role="alert"
+    >
+        <div class="message">
+            <span id="toast-message">
+                {{ title }}
+            </span>
+        </div>
+        <div class="buttons">
+            <button v-if="needRefresh" type="button" class="reload" @click="updateServiceWorker()">
+                Reload
+            </button>
+            <button type="button" @click="close">
+                Close
+            </button>
+        </div>
+    </div>
+</template>
+
+<style scoped>
```

Comment:
a lang=scss attribute should be added here

Reply:
I do not quite understand this. I tried adding the attribute but got the following error on build:
`Error: [vite-plugin-pwa:build] Preprocessor dependency "sass" not found. Did you install it? Try 'npm install -D sass.'`

From what I can observe, it functions properly in its absence, and the background variable appears to be integrating as expected. Could you provide further details?

Reply:
I agree with @cgjgh - there is n need for `lang=scss` here, it's using standard CSS variables

Reply:
My IDE was complaining about it, can be safely ignored